src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const setParent = useCallback((arrayOfPeople: Person[]): Person[] => {
+    return arrayOfPeople.map(person => {
+      const mother = arrayOfPeople.find(p => p.name === person.motherName);
+      const father = arrayOfPeople.find(p => p.name === person.fatherName);
+
+      return {
+        ...person,
+        mother,
+        father,
+      };
+    });
+  }, []);
```

Comment:
The `setParent` function is called for each person in the people array, and inside it, you are using the `find` method, which also loops through the array. This results in quadratic time complexity. You could optimize this by creating a map of people by name before calling `setParent`, and then just looking up the parents in the map. This would reduce the time complexity to linear.